resources/js/features/achievements/components/CreateAchievementTicketMainRoot/CreateAchievementTicketForm/useCreateAchievementTicketForm.ts

```diff
@@ -0,0 +1,96 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useMutation } from '@tanstack/react-query';
+import axios from 'axios';
+import { useLaravelReactI18n } from 'laravel-react-i18n';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+
+import { toastMessage } from '@/common/components/+vendor/BaseToaster';
+import { usePageProps } from '@/common/hooks/usePageProps';
+import { TicketType } from '@/common/utils/generatedAppConstants';
+
+const createAchievementTicketFormSchema = z.object({
+  /** @see TicketType.php */
+  issue: z.enum(['DidNotTrigger', 'TriggeredAtWrongTime', 'NetworkIssue']),
+  emulator: z.string().min(1),
+  emulatorVersion: z.string().optional(),
+  core: z.string().optional(),
+  mode: z.enum(['hardcore', 'softcore']),
+  hash: z.string().min(1),
+  description: z
+    .string()
+    .min(25, { message: 'Please be more detailed in your description.' })
+    .refine((desc) => !/(n'?t|not?).*(work|trigger)/i.test(desc), {
+      message: 'Please be more detailed in your description.',
+    })
+    .refine((desc) => !/(internet|wifi|connection|network)/i.test(desc), {
+      message: 'Do not open tickets for network issues.',
+    }),
+});
+
+export type CreateAchievementTicketFormValues = z.infer<typeof createAchievementTicketFormSchema>;
+
+export function useCreateAchievementTicketForm(
+  initialValues: Partial<CreateAchievementTicketFormValues>,
+) {
+  const {
+    achievement,
+    ziggy: { query },
+  } = usePageProps<App.Platform.Data.CreateAchievementTicketPageProps>();
+
+  const { t } = useLaravelReactI18n();
+
+  const form = useForm<CreateAchievementTicketFormValues>({
+    resolver: zodResolver(createAchievementTicketFormSchema),
+    defaultValues: initialValues,
+  });
+
+  const mutation = useMutation({
+    mutationFn: (formValues: CreateAchievementTicketFormValues) => {
+      return axios.post<{ message: string; ticketId: string }>(route('api.ticket.store'), {
+        ticketableModel: 'achievement',
+        ticketableId: achievement.id,
+        mode: formValues.mode,
+        issue: getTicketTypeFromIssue(formValues.issue),
+        description: formValues.description,
+        emulator: formValues.emulator,
+        emulatorVersion: formValues.emulatorVersion?.trim() ? formValues.emulatorVersion : null,
+        core: formValues.core,
+        gameHashId: Number(formValues.hash),
+        extra: query.extra ?? null,
+      });
+    },
+  });
+
+  const onSubmit = async (formValues: CreateAchievementTicketFormValues) => {
+    toastMessage.promise(mutation.mutateAsync(formValues), {
+      loading: t('Submitting...'),
+      success: (submitResponse) => {
+        setTimeout(() => {
+          const { ticketId } = submitResponse.data;
+
+          // TODO use router.visit after migrating this page to React
+          window.location.href = route('ticket.show', { ticket: ticketId });
+        }, 1000);
```

Comment:
Can this button be disabled while submitting? The delay here offers plenty of opportunity to click the button multiple times before the page turns.
![image](https://github.com/user-attachments/assets/a4c4e3fd-e4fc-468d-8f66-f37a9d4519ca)

This first time I did it because I wasn't sure I had actually clicked on it because the page turn wasn't immediate.

Reply:
Not a problem - updated in latest.